ui/src/BulkUploads.js

```diff
@@ -0,0 +1,336 @@
+    this.setState({
+      showDetails: true,
+      selectedJob: job.id,
+      selectedJobType: jobType,
+    });
+  };
+
+  handleClosedDetails = () => {
+    this.setState({ showDetails: false });
+  };
+
+  onProcessJob = () => {
+    fetch(`/api/job/${this.state.selectedJob}/process`, {
+      method: "POST",
+    });
+  };
+
+  onCancelJob = () => {
+    fetch(`/api/job/${this.state.selectedJob}/cancel`, {
+      method: "POST",
+    });
+  };
+
+  render() {
+    var selectedJob;
+    var detailsDialogTitle;
+    var loadPermission;
+    switch (this.state.selectedJobType) {
+      case "BULK_REFUSAL":
+        selectedJob = this.state.bulkRefusalJobs.find(
+          (job) => job.id === this.state.selectedJob
+        );
+        detailsDialogTitle = "Bulk Refusal Detail";
+        loadPermission = "LOAD_BULK_REFUSAL";
+        break;
+      default:
+    }
+
+    return (
+      <div style={{ padding: 20 }}>
+        <Link
+          to={`/collex?surveyId=${this.props.surveyId}&collexId=${this.props.collectionExerciseId}`}
+        >
+          ← Back to collection exercise details
+        </Link>
+        <Typography variant="h4" color="inherit">
+          Uploaded Bulk Process Files
+        </Typography>
+        {this.buildBulkProcessTable(
+          this.state.bulkRefusalJobs,
+          "BULK_REFUSAL",
+          "Bulk Refusals",
+          "LOAD_BULK_REFUSAL",
+          "VIEW_BULK_REFUSAL_PROGRESS"
+        )}
+        <JobDetails
+          jobTitle={detailsDialogTitle}
+          job={selectedJob}
+          showDetails={this.state.showDetails}
+          handleClosedDetails={this.handleClosedDetails}
+          onClickAway={this.handleClosedDetails}
+          onProcessJob={this.onProcessJob}
+          onCancelJob={this.onCancelJob}
+          authorisedActivities={this.state.authorisedActivities}
+          loadPermission={loadPermission}
+        ></JobDetails>
+      </div>
+    );
+  }
+
+  buildBulkProcessTable(
+    bulkJobs,
+    jobType,
+    jobTitle,
+    loadPermission,
+    viewerPermission
+  ) {
+    const bulkJobTableRows = bulkJobs.map((job, index) => (
+      <TableRow key={job.createdAt}>
+        <TableCell component="th" scope="row">
+          {job.fileName}
+        </TableCell>
+        <TableCell>{job.createdAt}</TableCell>
+        <TableCell align="right">
+          <Button
+            onClick={() => this.handleOpenDetails(job, jobType)}
+            variant="contained"
+          >
+            {convertStatusText(job.jobStatus)}{" "}
+            {[
+              "STAGING_IN_PROGRESS",
+              "VALIDATION_IN_PROGRESS",
+              "PROCESSING_IN_PROGRESS",
+            ].includes(job.jobStatus) && (
+              <CircularProgress size={15} style={{ marginLeft: 10 }} />
+            )}
+          </Button>
+        </TableCell>
+      </TableRow>
+    ));
+
+    return (
+      <div style={{ padding: 20 }}>
+        {this.state.authorisedActivities.includes(viewerPermission) && (
```

Comment:
might as well do this check right at the start of the `buildBulkProcessTable` and return an empty/null fragment if they don't have permission... save on processing time